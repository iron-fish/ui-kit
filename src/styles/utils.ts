import { BASE_FONT_SIZE } from 'src/styles/constants'

export const asPx = (x: string) =>
  `${parseFloat(x.slice(0, -3)) * BASE_FONT_SIZE}px`
