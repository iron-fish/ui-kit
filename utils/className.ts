import bemClassName from 'bem-classname'

export const bem = baseName => bemClassName.bind(null, baseName)
