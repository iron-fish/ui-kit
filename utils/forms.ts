/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

import { ChangeEvent, Dispatch, SetStateAction } from 'react'

// naive validators

// reusable default value
export const UNSET = ''

// emails should have an @ sign and dot that isn't at the end
export const validateEmail = (x: string) => {
  const dot = x.indexOf('.')
  return x.indexOf('@') > 0 && dot > 0 && dot !== x.length - 1
}

// non-zero width strings
export const exists = (x: string) => x.trim().length > 0

// reusable error text
export const defaultErrorText = `This field is required`

// take a setter and turn it into an event handler which sets
export function setStateOnChange(setter: Dispatch<SetStateAction<string>>) {
  return (e: ChangeEvent) => {
    setter((e.target as HTMLInputElement).value)
  }
}
