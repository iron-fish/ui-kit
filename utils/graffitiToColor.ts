/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

const fishAvatarColors = [
  '#FFAFAF',
  '#94ED6B',
  '#A3E9FF',
  '#7657CE',
  '#E7B453',
  '#54FF17',
  '#1D4423',
  '#D46161',
  '#AFF1FF',
  '#F1CB00',
]

export function graffitiToColor(graffiti: string): string {
  // djb2 (xor)
  let hash = 5381

  for (let i = 0; i < graffiti.length; i++) {
    hash = (hash * 33) ^ graffiti.charCodeAt(i)
  }

  return fishAvatarColors[(hash >>> 0) % fishAvatarColors.length]
}
