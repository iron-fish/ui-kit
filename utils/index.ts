export * from './forms'
export * from './graffitiToColor'
export * from './numberToOrdinal'
export * from './scroll'
