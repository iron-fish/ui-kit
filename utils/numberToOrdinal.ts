/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

/**
 * Adds an ordinal suffix to a given number, e.g. 1 -> 1st
 */
export function numberToOrdinal(num: number): string {
  // From https://stackoverflow.com/a/13627586
  const j = num % 10
  const k = num % 100

  if (j === 1 && k !== 11) {
    return `${num}st`
  }
  if (j == 2 && k != 12) {
    return `${num}nd`
  }
  if (j == 3 && k != 13) {
    return `${num}rd`
  }
  return `${num}th`
}
