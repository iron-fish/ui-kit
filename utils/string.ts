export const capitalize = (z: string) =>
  z.slice(0, 1).toUpperCase() + z.slice(1, Infinity).toLowerCase()
