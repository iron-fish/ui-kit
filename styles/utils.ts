import { BASE_FONT_SIZE } from 'styles/constants'

export const asPx = (x: string) =>
  `${parseFloat(x.slice(0, -3)) * BASE_FONT_SIZE}px`
