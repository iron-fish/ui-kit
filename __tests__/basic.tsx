import { FC } from 'react'
// import { render } from '@testing-library/react'
import { chakra } from '@chakra-ui/react'
import '@testing-library/jest-dom'

import { render } from 'jest.utils'

const Basic: FC = () => <chakra.strong>cool</chakra.strong>

test('smoke test', () => {
  // expect(COMPONENTS).toBeTruthy()
  const rendered = render(<Basic />)
  expect(rendered.container).toMatchSnapshot()
})
