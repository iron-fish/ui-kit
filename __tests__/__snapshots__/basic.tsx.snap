// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke test 1`] = `
<div>
  <strong
    class="css-0"
  >
    cool
  </strong>
</div>
`;
